metabrainz.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+"""
+Author : Salvador <salvador.estrella.ortiz@gmail.com>
+Date   : 2022-09-30
+Purpose: Query Metadata from MusicBrainz API
+"""
+
+import argparse
+import requests
+from pprint import pprint
+
+
+# --------------------------------------------------
+def get_args():
+    """Get command-line arguments"""
+
+    parser = argparse.ArgumentParser(
+        description="Retrieve metadata from MusicBrainz API",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "-a",
+        "--artist",
+        help="An artist to search for",
+        metavar="str",
+        type=str,
+        nargs="+",
+        required=True,
+    )
+
+    return parser.parse_args()
+
+
+# --------------------------------------------------
+class MusicMetadata:
+    artist_endpoit = "http://musicbrainz.org/ws/2/artist/"
+    release_groups_endpoint = "http://musicbrainz.org/ws/2/release-group/"
+    releases_endpoint = "http://musicbrainz.org/ws/2/release/"
+
+    def __init__(self, artist: list[str]):
+        self.artist = artist
+
+    def pick_artist(self, artist_list):
+        for artist in artist_list:
+            if artist["score"] == 100:
```

Comment:
**suggestion:** Leave a `TODO` comment to remember that you need to deal with artist names that don't return a 100 score result.

Alternatively raise a not implemented error whenever this case happens. Something like

```python
def pick_artists(artists: List[str]):
    artist = None
    for possible_artist in artists:
        if not artists:
            artist = possible_artist
        elif artist["score"] < possible_artist["score"]:
            artist = possible_artist
    if artist and artist["score"] != 100:
        raise raise NotImplementedError("Selection of other artists with lower score than 100 not implemented yet")
    return artist
```

Reply:
Changed the class method to look like the one suggested. 

**suggestion**: Only deal with artist that have a score of 100 or raise IndexError/ValueError for non existing match.

musicbrainz API will return an empty list of artists if no close match found, otherwise it will return a list of artist and automatically assign a score of 100 to the first result. Here's an example result when querying "coldplai"
```python
(Pdb) l
 47             artist = "%20".join(artist)
 48             artist_query = f"{cls.artist_endpoit}?query={artist}&fmt=json"
 49             response = requests.get(artist_query)
 50             artist_list = response.json()["artists"]
 51             breakpoint()
 52  ->         return artist_list

(Pdb) pp(response.json())
{'artists': [], 'count': 0, 'created': '2022-10-07T16:51:14.545Z', 'offset': 0}
```
And here's the response for querying "tucanes":
```python
(Pdb) pp(response.json())
{'artists': [{'id': 'f1802d84-5885-4cc0-81c6-c4823510ebf4',
              'life-span': {'ended': None},
              'name': 'Los Tucanes de Tijuana',
              'score': 100,
              'sort-name': 'Tucanes de Tijuana, Los',
              'tags': [{'count': 1, 'name': 'corrido'},
                       {'count': 1, 'name': 'norteño'},
                       {'count': 1, 'name': 'cumbias'},
                       {'count': 1, 'name': 'regional mexicano'},
                       {'count': 1, 'name': 'banda'},
                       {'count': 1, 'name': 'ballad'}],
              'type': 'Group',
              'type-id': 'e431f5f6-b5d2-343d-8b36-72607fffb74b'},
             {'area': {'id': '01448ddc-6ee3-3fa4-b136-507d984e31ee',
                       'life-span': {'ended': None},
                       'name': 'Guatemala',
                       'sort-name': 'Guatemala',
                       'type': 'Country',
                       'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de'},
              'begin-area': {'id': '01448ddc-6ee3-3fa4-b136-507d984e31ee',
                             'life-span': {'ended': None},
                             'name': 'Guatemala',
                             'sort-name': 'Guatemala',
                             'type': 'Country',
                             'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de'},
              'country': 'GT',
              'id': '15da4d9d-38fd-4845-9f69-18fca559ee3a',
              'life-span': {'begin': '1995-05-12', 'ended': None},
              'name': 'Internacionales Tucanes De Guatemala',
              'score': 90,
              'sort-name': 'Internacionales Tucanes De Guatemala',
              'type': 'Orchestra',
              'type-id': 'a0b36c92-3eb1-3839-a4f9-4799823f54a5'}],
 'count': 2,
 'created': '2022-10-07T17:00:13.710Z',
 'offset': 0}
```